src/function/table/arrow_conversion.cpp

```diff
@@ -341,6 +346,170 @@ static void IntervalConversionMonthDayNanos(Vector &vector, ArrowArray &array, A
+
+	UnifiedVectorFormat run_end_format;
+	UnifiedVectorFormat value_format;
+	runs.ToUnifiedFormat(compressed_size, run_end_format);
+	values.ToUnifiedFormat(compressed_size, value_format);
+	auto run_ends_data = run_end_format.GetData<RUN_END_TYPE>(run_end_format);
+	auto values_data = value_format.GetData<VALUE_TYPE>(value_format);
+	auto result_data = FlatVector::GetData<VALUE_TYPE>(result);
+	auto &validity = FlatVector::Validity(result);
+
+	// According to the arrow spec, the 'run_ends' array is always valid
+	// so we will assume this is true and not check the validity map
+
+	// Now construct the result vector from the run_ends and the values
+
+	idx_t &run = run_end_encoding.run_index;
+	// FIXME: can not support array.offset currently because
+	// `state.chunk_offset` and `run_end_encoding.run_index` are assumed to be in sync
+	// If we add `array.offset` to the mix, we need to recalculate the `run_index`
+	idx_t logical_index = scan_offset;
+	idx_t index = 0;
+	if (value_format.validity.AllValid()) {
+		// None of the compressed values are NULL
+		for (; run < compressed_size; run++) {
+			auto run_end_index = run_end_format.sel->get_index(run);
+			auto value_index = value_format.sel->get_index(run);
+			auto &value = values_data[value_index];
+			auto run_end = static_cast<idx_t>(run_ends_data[run_end_index]);
+
+			D_ASSERT(run_end > (logical_index + index));
+			auto to_scan = run_end - (logical_index + index);
+			// Cap the amount to scan so we don't go over size
+			to_scan = MinValue<idx_t>(to_scan, (count - index));
+
+			for (idx_t i = 0; i < to_scan; i++) {
+				result_data[index + i] = value;
+			}
+			index += to_scan;
+			if (index >= count) {
+				if (logical_index + index >= run_end) {
+					// The last run was completed, forward the run index
+					run++;
+				}
+				break;
+			}
+		}
+	} else {
+		for (; run < compressed_size; run++) {
+			auto run_end_index = run_end_format.sel->get_index(run);
+			auto value_index = value_format.sel->get_index(run);
+			auto run_end = static_cast<idx_t>(run_ends_data[run_end_index]);
+
+			D_ASSERT(run_end > (logical_index + index));
+			auto to_scan = run_end - (logical_index + index);
+			// Cap the amount to scan so we don't go over size
+			to_scan = MinValue<idx_t>(to_scan, (count - index));
+
+			if (value_format.validity.RowIsValidUnsafe(value_index)) {
+				auto &value = values_data[value_index];
+				for (idx_t i = 0; i < to_scan; i++) {
+					result_data[index + i] = value;
+					validity.SetValid(index + i);
+				}
+			} else {
+				for (idx_t i = 0; i < to_scan; i++) {
+					validity.SetInvalid(index + i);
+				}
+			}
+			index += to_scan;
+			if (index >= count) {
+				if (logical_index + index >= run_end) {
+					// The last run was completed, forward the run index
+					run++;
+				}
+				break;
+			}
+		}
+	}
+}
+
+template <class RUN_END_TYPE>
+static void FlattenRunEndsSwitch(Vector &result, ArrowRunEndEncodingState &run_end_encoding, idx_t compressed_size,
```

Comment:
Yea it can, that's why there's a NotImplementedException

I figured for now we should support integers because it makes no sense to do this for other types imo

Reply:
I figured dictionary would make more sense, but thinking about it longer than 5 seconds yea it makes sense to support other types.